pallets/committee/src/lib.rs

```diff
@@ -353,24 +361,33 @@ pub mod pallet {
 		/// meeting voting requirements can be executed.
 		#[pallet::weight((T::WeightInfo::vote(), DispatchClass::Operational))]
 		pub fn vote(origin: OriginFor<T>, proposal_hash: HashFor<T>, vote: VoteKind) -> DispatchResult {
-			// Only members can vote
 			let voter = Self::ensure_member(origin)?;
-
-			Votes::<T>::try_mutate(&proposal_hash, |votes| {
-				if let Some(votes) = votes {
-					// Can only vote within the allowed range of blocks for this proposal
-					ensure!(Self::within_voting_period(votes), Error::<T>::NotInVotingPeriod);
-					// members can vote only once
-					ensure!(!votes.has_voted(&voter.account_id), Error::<T>::DuplicateVote);
-					votes.cast_vote(MemberVote::new(voter.clone(), vote.clone())); // mutates votes in place
-					Self::deposit_event(Event::VoteCast(voter, proposal_hash, vote));
-					Ok(())
-				} else {
-					Err(Error::<T>::NoProposalWithHash)
+			VotingTimeouts::<T>::try_mutate(&voter.account_id, |maybe_timeout| -> DispatchResult {
+				if maybe_timeout
+					.take()
+					.filter(|block_number| {
+						frame_system::Pallet::<T>::block_number().saturating_sub(*block_number) < T::VotingPeriod::get()
+					})
+					.is_some()
+				{
```

Comment:
this means members included at genesis are missing in the `VotingTimeouts` set, right?
I wonder if we should instead make this conistent for all members in `VotingTimeouts`, so that we track the first block at which they are eligible to vote, which is `0` for members in genesis

Reply:
This can make all members equal at the `VotingEligibility` part in logic, but seems have no benefits for solving the voting attack problem since this check if for new members

Reply:
Yes, this is just for consistency's sake.